pkg/accounting/accounting.go

```diff
@@ -219,3 +226,34 @@ func (a *Accounting) getPeerBalance(peer swarm.Address) (*PeerBalance, error) {
 func (pb *PeerBalance) freeBalance() int64 {
 	return pb.balance - int64(pb.reserved)
 }
+
+func (a *Accounting) getPeersBalances() (map[string]int64, error) {
+	peersBalances := make(map[string]int64)
+
+	for peer, balance := range a.balances {
+		peersBalances[peer] = balance.balance
+	}
+
+	err := a.store.Iterate("balance_", func(key, val []byte) (stop bool, err error) {
+
+		k := string(key)
+		split := strings.SplitAfter(k, "balance_")
+		if len(split) != 2 {
+			return false, fmt.Errorf("invalid overlay key: %s", k)
+		}
+
+		addr, err := swarm.ParseHexAddress(split[1])
+
+		if err != nil {
+			return false, err
+		}
+
+		if val, ok := peersBalances[addr.String()]; !ok {
+			peersBalances[addr.String()] = val
+		}
+
+		return false, nil
+	})
+
+	return peersBalances, err
```

Comment:
probably yes, as the error could be from the statestore operation, however, before that, we already read items from memory that are probably worth returning.

Reply:
> probably yes, as the error could be from the statestore operation, however, before that, we already read items from memory that are probably worth returning.

so basically what you're saying is that the calling context should ignore the error in case there was one (since you are assuming values should still be inspected although an error has occurred). i second what marcello is writing here and would vote to have an explicit error check returning a `nil, err` if there was an error. otherwise `peerBalances, nil`